scripts/S3/generate-s3cfg.sh

```diff
@@ -1,11 +1,20 @@
 #!/bin/bash
 
+set -eu -o pipefail
+
 # Simple script for generating a sample s3cfg for a couple of known
 # and tested S3 providers.
 
 function usage() {
     echo "Usage:" 1>&2
-    echo "  $0 aws|exoscale|safespring|citycloud access_key secret_key host_base region" 1>&2
+    echo "  $0 {aws|exoscale|safespring|citycloud} {access_key} {secret_key} {host_base} [region]" 1>&2
+    echo "  host_base - the host (and port if other than default) of the service" 1>&2
+    echo "  region - the location where the buckets should be stored. This is ignored for exoscale, safespring and citycloud." 1>&2
+    echo "Examples:" 1>&2
+    echo "  $0 aws abc 123 s3.amazonaws.com eu-north-1" 1>&2
```

Comment:
The AWS one is a bit weird as the endpoint isn't used but it's fine in order to keep the usage simple.